### PR TITLE
Support specifying inverted index columns when creating table

### PR DESCRIPTION
**Pull request by wubiaoi**
For #1865

### CODE REVIEW

File: fe/src/main/java/org/apache/doris/catalog/OlapTable.java [hunk 1]
```diff
@@ -844,6 +864,17 @@ public void write(DataOutput out) throws IOException {
             Text.writeString(out, colocateGroup);
         }
 
+        // inverted index columns
```
Comment (kangkaisen) [inline fe/src/main/java/org/apache/doris/catalog/OlapTable.java]: Write invertedIndexColumns should after baseIndexId.

File: fe/src/main/java/org/apache/doris/catalog/OlapTable.java [hunk 2]
```diff
@@ -933,6 +964,16 @@ public void readFields(DataInput in) throws IOException {
             }
         }
 
+        if (Catalog.getCurrentCatalogJournalVersion() >= FeMetaVersion.VERSION_64) {
```
Comment (kangkaisen) [inline fe/src/main/java/org/apache/doris/catalog/OlapTable.java]: should after baseIndexId.

Comment (kangkaisen) [inline fe/src/main/java/org/apache/doris/catalog/OlapTable.java]: should after baseIndexId.

**Comment by kangkaisen** [comment]
Please add UT and doc.

**Comment by kangkaisen** [comment]
@wubiaoi  Hi, The schema change for invert index In BE is another PR?

File: fe/src/main/java/org/apache/doris/common/util/PropertyAnalyzer.java [hunk 3]
```diff
@@ -371,4 +377,38 @@ public static long analyzeTimeout(Map<String, String> properties, long defaultTi
...
+        String idxColsStr = properties.get(PROPERTIES_INVERTED_INDEX_COLUMNS);
+        properties.remove(PROPERTIES_INVERTED_INDEX_COLUMNS);
+        if (Strings.isNullOrEmpty(idxColsStr)) {
+            return Sets.newHashSet();
+        }
+        Set<String> resultCols = Sets.newHashSet();
+        Map<String, Column> columnMap = columns.stream()
+                .collect(Collectors.toMap(col -> col.getName().toLowerCase(), Function.identity()));
+
+        List<String> idxCols = COMMA_SPLITTER.splitToList(idxColsStr.toLowerCase());
```
Comment (kangpinghuang) [inline fe/src/main/java/org/apache/doris/common/util/PropertyAnalyzer.java]: I think this List is name, not idx, So I think colNames is a better name.

File: fe/src/main/java/org/apache/doris/common/util/PropertyAnalyzer.java [hunk 4]
```diff
@@ -371,4 +377,38 @@ public static long analyzeTimeout(Map<String, String> properties, long defaultTi
...
+        for (String idxCol : idxCols) {
+            if (columnMap.containsKey(idxCol)) {
+                Column column = columnMap.get(idxCol);
+                if (column.isKey()) {
```
Comment (kangpinghuang) [inline fe/src/main/java/org/apache/doris/common/util/PropertyAnalyzer.java]: here,
if the table is a agg table, inverted index can only be used in key columns, not value columns;
if the table is duplicate table, inverted index can be used in both key and value;
if the table is unique table, inverted index can only be used in key column.

File: fe/src/main/java/org/apache/doris/catalog/Catalog.java [hunk 5]
```diff
@@ -3852,6 +3866,13 @@ public static void getDdlStmt(Table table, List<String> createTableStmt, List<St
                 sb.append(colocateTable).append("\"");
             }
 
+            // 6. inverted index
+            Set<String> invertedIdxCols = olapTable.getInvertedIndexColumns();
+            if (invertedIdxCols != null) {
+                sb.append(",\n \"").append(PropertyAnalyzer.PROPERTIES_INVERTED_INDEX_COLUMNS).append("\" = \"");
+                sb.append(Joiner.on(", ").join(invertedIdxCols)).append("\"");
+            }
```
Comment (kangpinghuang) [inline fe/src/main/java/org/apache/doris/catalog/Catalog.java]: add a UT to test this logic?

**Review by kangpinghuang — CHANGES_REQUESTED** [review]
There are some problems.

**Comment by wubiaoi** [comment]
`CREATE INDEX` and `DROP INDEX`  are more reasonable

**Pull request closed without merging**